### PR TITLE
sql: attempt to deflake TestRaceWithIndexBackfillMerge

### PR DESCRIPTION
**Pull request by ajwerner**
See [here](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests_BazelUnitTests/6186518?buildTab=overview&showRootCauses=false&expandBuildProblemsSection=true&expandBuildTestsSection=true&expandBuildChangesSection=true&expandBuildDeploymentsSection=true#%2Fbazel-testlogs%2Fpkg%2Fsql%2Fsql_test%2Fshard_11_of_16)

Release justification: testing only change

Release note: None

### CODE REVIEW

**Comment by ajwerner** [comment]
This is fallout from https://github.com/cockroachdb/cockroach/pull/82758.

File: pkg/sql/mvcc_backfiller_test.go [hunk 1]
```diff
@@ -700,7 +701,11 @@ func splitIndex(
 		}
 
 		for _, target := range rkt.KT.Targets {
-			if err := tc.TransferRangeLease(desc, target); err != nil {
+			if err := retry.WithMaxAttempts(context.Background(), retry.Options{
+				MaxBackoff: time.Millisecond,
```
Comment (stevendanna) [inline pkg/sql/mvcc_backfiller_test.go:705]: 🤷 No idea what is reasonable here.

**Review by stevendanna — APPROVED** [review]
👍 LGTM. Thanks for looking into this.

**Comment by ajwerner** [comment]
TFTR!

bors r=stevendanna

**Pull request merged**